spec/index.bs

```diff
@@ -24,12 +24,12 @@ typedef unsigned long u32;
 typedef unsigned long long u64;
 
 dictionary GPUColorDict {
-    required float r;
-    required float g;
-    required float b;
-    required float a;
+    required double r; // `double` is large enough for precise i32/u32.
+    required double g;
+    required double b;
+    required double a;
 };
-typedef (sequence<float> or GPUColorDict) GPUColor;
+typedef (sequence<double> or GPUColorDict) GPUColor;
```

Comment:
did you consider the alternative of explicitly supporting integers, i.e
```
dictionary GPUColorFloatDict {
    required float r;
    required float g;
    required float b;
    required float a;
};
dictionary GPUColorIntDict {
    required i32 r;
    required i32 g;
    required i32 b;
    required i32 a;
};
typedef (sequence<float> or GPUColorFloatDict or sequence<i32> or GPUColorIntDict) GPUColor;
```

It seems a bit strange to try to fit integers into doubles if we already have `GPUColor` defined as an enum (of sequence or a dict).

Reply:
IIRC that's not valid; the bindings resolution rules can't choose an overload based based on the type. If an app passes [4,4,4,4], the bindings can't choose which C++ overload to call (the int one or the float one).